### DISCUSSION TITLE
Allow "ForceAuthn" in the SAML AuthnRequest to be configurable as True or False

### DISCUSSION THREAD

**Discussion by vsamtani** [latchset/mod_auth_mellon#83] [Ideas]
Currently the AuthnRequest attribute ForceAuthn is always "false". Could that be determined by a configuration directive instead?

https://github.com/latchset/mod_auth_mellon/blob/d5cfa396f69ba2ee0d1048fe59c2f0f4a6d077ef/auth_mellon_handler.c#L3017

**Comment by BriceDuchatel** [comment]
Oh yes ! It would be really good!

**Comment by thijskh** [comment]
This would of course be possible to allow someone to configure.

However I'd like to caution against using this option as it's far from a robust security measure, although it suggests it is. ForceAuthn cannot really be enforced, it's just a hint mostly. As mod_auth_mellon allows unsolicited (idp-first) SSO, there's a trivial way around this security measure in many cases if you want to.

**Reply by vsamtani** [reply]
Agreed, it's not a solid security measure. It's more about UX. There are use cases where the SP owner wants to request user-interactive authentication at the IDP, with the caveat that it could be overridden. Shibboleth permits this in configuration, but we can't swap out Shib for mod_auth_mellon in those use cases unless we can replicate the behaviour.

**Reply by thijskh** [reply]
A pull request would be welcome, I'd recommend to add a disclaimer along these lines in the documentation of the config option.

**Reply by vsamtani** [reply]
Sadly, I'm an interested user, not a developer, so I hope someone with the right skills can pick this up and make it happen.... :)

**Reply by kubik256** [reply]
Any update on this topic?
Every other SAML integration have this option :-(
